lightblue/src/main/java/org/esbtools/eventhandler/lightblue/BulkLightblueRequester.java

```
@@ -0,0 +1,158 @@
+    private final LightblueClient lightblue;
+    private final Map<LazyResult, AbstractLightblueDataRequest[]> queuedRequests =
+            Collections.synchronizedMap(new HashMap<>());
+
+    public BulkLightblueRequester(LightblueClient lightblue) {
+        this.lightblue = lightblue;
+    }
+
+    @Override
+    public LightblueResponsePromise queueRequests(AbstractLightblueDataRequest... requests) {
+        return new BulkResponsePromise(requests);
+    }
+
+    private void doQueuedRequestsAndPopulateResults() throws LightblueException {
+        Map<LazyResult, AbstractLightblueDataRequest[]> batch;
+
+        synchronized (queuedRequests) {
+            batch = new HashMap<>(queuedRequests);
+            queuedRequests.clear();
+        }
+
+        DataBulkRequest bulkRequest = new DataBulkRequest();
+
+        for (AbstractLightblueDataRequest[] allRequestBatches : batch.values()) {
+            for (AbstractLightblueDataRequest requestInBatch : allRequestBatches) {
+                bulkRequest.add(requestInBatch);
+            }
+        }
+
+        LightblueBulkDataResponse bulkResponse = lightblue.bulkData(bulkRequest);
+
+        for (Entry<LazyResult, AbstractLightblueDataRequest[]> lazyResultToRequests : batch.entrySet()) {
+            LazyResult lazyResult = lazyResultToRequests.getKey();
+            AbstractLightblueDataRequest[] requests = lazyResultToRequests.getValue();
+            Map<AbstractLightblueDataRequest, LightblueDataResponse> responseMap = new HashMap<>(requests.length);
+
+            for (AbstractLightblueDataRequest request : requests) {
+                LightblueResponse response = bulkResponse.getResponse(request);
+
+                if (response instanceof LightblueErrorResponse) {
+                    LightblueErrorResponse errorResponse = (LightblueErrorResponse) response;
+                    if (errorResponse.hasLightblueErrors() || errorResponse.hasDataErrors()) {
+                        // TODO: handle errors
```

Instead of just continuing, can we just throw immediately? I'd rather have subpar behavior that bad behavior (errors are ignored here for now, no?)